File: projects/punk0/code/increment.py
# increment the first of a list
#   [1,2,3,4]
# returns:
#   [2,2,3,4]
def increment(x):
    r = []

    for i in range(len(x)):
        v = x[i]
        if i == 0:
            v += 1
        r.append(v)

    return r


print(increment([1, 1, 2, 3, 3, 4, 1, 2, 7, 1]))

File: projects/punk0/code/increment.go
package main

import (
	"fmt"
)

// increment the first of a list
//   [1,2,3,4]
// returns:
//   [2,2,3,4]
func increment(x []int) []int {
	r := []int{}

	for i, v := range x {
		if i == 0 {
			v++
		}
		r = append(r, v)
	}

	return r
}

func main() {
	fmt.Printf("%v\n", increment([]int{1, 1, 2, 3, 3, 4, 1, 2, 7, 1}))
}
